source/Octostache/Templates/Functions/TextSubstringFunction.cs

```diff
@@ -16,12 +16,17 @@ static class TextSubstringFunction
             if (options.Length == 1 && int.Parse(options[0]) > argument.Length)
                 return null;
 
-            if (options.Length == 2 && int.Parse(options[0]) + int.Parse(options[1]) > argument.Length)
+            if (options.Length == 2 && int.Parse(options[0]) > argument.Length)
                 return null;
 
-            return argument.Substring(
-                options.Length == 1 ? 0 : int.Parse(options[0]),
-                options.Length == 1 ? int.Parse(options[0]) : int.Parse(options[1]));
+            var startIndex = options.Length == 1 ? 0 : int.Parse(options[0]);
+            var length = options.Length == 1 ? int.Parse(options[0]) : int.Parse(options[1]);
+
+            // If starting position is valid but the length would exceed the string, use the remaining length of the string
+            if (startIndex < argument.Length && startIndex + length > argument.Length)
```

Comment:
Do we clamp the inputs to non-negative integers? It's probably a case of stupid inputs get stupid outputs, but is it worth handling if the user specifies a negative `startIndex`?